picard/util/__init__.py

```diff
@@ -1168,10 +1174,11 @@ def strxfrm(string):
 }
 
 
-def detect_unicode_encoding(path):
-    """Attempts to guess the unicode encoding of a file based on the BOM.
+def detect_file_encoding(path, read_limit=1024*256):
+    """Attempts to guess the unicode encoding of a file based on the BOM, and
+    depending on avalibility, using a charset detection method.
 
-    Assumes UTF-8 by default if there is no BOM.
+    Assumes UTF-8 by default if no other encoding is detected.
 
     Args:
         path: The path to the file
```

Comment:
Please document new `read_limit` here.

BTW, I wonder if a name likes `max_bytes_to_read` would be better or not. What do you think?

Reply:
I think `max_bytes_to_read` will be much better variable name. Adding it.